src/base/torrentfilter.cpp

```diff
@@ -121,11 +126,24 @@ bool TorrentFilter::setCategory(const QString &category)
     return false;
 }
 
+bool TorrentFilter::setTag(const QString &tag)
+{
+    // QString::operator==() doesn't distinguish between empty and null strings.
+    if ((m_tag != tag)
+        || (m_tag.isNull() && !tag.isNull())
```

Comment:
See comment in the code above. The problem is `QString() == QString("")` is true.